tests/dump_tests/module_tests/copp_test.py

```diff
@@ -0,0 +1,215 @@
+import json
+import os
+import sys
+import jsonpatch
+import unittest
+import pytest
+from deepdiff import DeepDiff
+from mock import patch
+from dump.helper import create_template_dict, sort_lists
+from dump.plugins.copp import Copp
+from .mock_sonicv2connector import MockSonicV2Connector
+
+module_tests_path = os.path.dirname(__file__)
+dump_tests_path = os.path.join(module_tests_path, "../")
+tests_path = os.path.join(dump_tests_path, "../")
+dump_test_input = os.path.join(tests_path, "dump_input")
+
+# Location for dedicated db's used for UT
+copp_files_path = os.path.join(dump_test_input, "copp")
+
+dedicated_dbs = {}
+dedicated_dbs['CONFIG_DB'] = os.path.join(copp_files_path, "config_db.json")
+dedicated_dbs['APPL_DB'] = os.path.join(copp_files_path, "appl_db.json")
+dedicated_dbs['ASIC_DB'] = os.path.join(copp_files_path, "asic_db.json")
+dedicated_dbs['STATE_DB'] = os.path.join(copp_files_path, "state_db.json")
+
+
+def mock_connector(namespace, use_unix_socket_path=True):
+    return MockSonicV2Connector(dedicated_dbs, namespace=namespace, use_unix_socket_path=use_unix_socket_path)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def verbosity_setup():
+    print("SETUP")
+    os.environ["VERBOSE"] = "1"
+    yield
+    print("TEARDOWN")
+    os.environ["VERBOSE"] = "0"
+
+
+@patch("dump.match_infra.SonicV2Connector", mock_connector)
+@patch("dump.plugins.copp.Copp.CONFIG_FILE", os.path.join(dump_test_input, "copp_cfg.json"))
+class TestCoppModule(unittest.TestCase):
+
+    def test_usr_cfg_trap_and_copp_cfg_file_grp(self):
+        '''
+        Scenario: A custom COPP_TRAP table entry is defined by the user and the relevant Trap Group is configured through the copp_cfg file
+        '''
+        params = {}
+        params[Copp.ARG_NAME] = "snmp"
+        params["namespace"] = ""
```

Comment:
Can you add some test cases where namespace non-empty.

Reply:
Multi-asic tests are already covered in the match_infra tests [LINK](https://github.com/Azure/sonic-utilities/blob/master/tests/dump_tests/match_engine_test.py#L217) and also in the tests which verify the dump state tests which verify the "dump state cli" output. [LINK](https://github.com/Azure/sonic-utilities/blob/42664848239b4d41e28b86e86e79da532c292eef/tests/dump_tests/dump_state_test.py#L188) .

These UT's effectively capture the capability at both the match_infra and the top level cli levels. 

I don't think, adding non-default namespace tests here wouldn't add any value as it doesn't help verify anything specific to copp module implementation.